x/bet/types/bet.go

```diff
@@ -30,3 +31,50 @@ func NewBetFulfillment(
 		PayoutProfit:       payoutProfit,
 	}
 }
+
+// CheckSettlementEligiblity checks status of bet. It returns an error if
+// bet is canceled or settled already.
+func (bet *Bet) CheckSettlementEligiblity() error {
+	switch bet.Status {
+	case Bet_STATUS_SETTLED:
+		return ErrBetIsSettled
+	case Bet_STATUS_CANCELED:
+		return ErrBetIsCanceled
+	}
+
+	return nil
+}
+
+// SetResult sets the bet object results according to the market resolusion.
+func (bet *Bet) SetResult(market *markettypes.Market) error {
```

Comment:
do we need pointer to market here?

Reply:
Yes, Otherwise we need to pass the bet object and refill it by the caller.

Reply:
I am talking about the pointer to market